File: packages/essential_android_sdk/samples/QuickStart.kt
package io.essential.sdk.android.samples

import android.content.Context
import io.essential.sdk.android.EssentialClient
import io.essential.sdk.android.EssentialGenerateRequest
import io.essential.sdk.android.EssentialModelRequirement
import io.essential.sdk.android.EssentialServiceConfiguration
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.launch

fun connectAndStream(
    context: Context,
    scope: CoroutineScope,
    render: (String) -> Unit,
) {
    val configuration = EssentialServiceConfiguration(
        servicePackage = "com.example.essential_flutter",
        serviceClassName = "com.example.essential_flutter.service.EssentialService",
    )

    scope.launch {
        val client = EssentialClient.connect(context, configuration)
        client.generateStream(
            EssentialGenerateRequest(
                prompt = "端末上で要約して",
                modelRequirement = EssentialModelRequirement.fallback("essential-mini"),
                timeoutMs = 10_000,
            ),
        ).collect { chunk ->
            render(chunk.accumulatedText)
        }
    }
}

File: packages/essential_sdk_dart/example/quick_start.dart
import 'package:essential_sdk_dart/essential_sdk_dart.dart';

Future<void> main() async {
  final client = await EssentialClient.initialize(
    const EssentialConfiguration(
      defaultModelId: 'essential-mini',
      installedModels: <EssentialInstalledModel>[
        EssentialInstalledModel(
          modelId: 'essential-mini',
          modelPath: '/tmp/essential-mini.gguf',
          family: 'llama.cpp',
          contextWindow: 4096,
        ),
      ],
    ),
  );

  await for (final chunk in client.generateStream(
    const EssentialGenerateRequest(
      prompt: '端末内だけで応答して',
      modelRequirement: EssentialModelRequirement.fallback('essential-mini'),
      timeoutMs: 5000,
    ),
  )) {
    print(chunk.accumulatedText);
  }

  await client.dispose();
}